### PR TITLE
RichTextData: use a private property

### PR DESCRIPTION
**Pull request by jsnajdr**
In `RichTextData`, use a private class instance property for the private data, instead of a symbol-indexed public property. Private properties are now standard, they should work.

### CODE REVIEW

File: packages/rich-text/src/create.js [hunk 1]
```diff
@@ -138,22 +137,15 @@ export class RichTextData {
...
+		return getTextContent( this.#value );
 	}
 	// We could expose `toHTMLElement` at some point as well, but we'd only use
 	// it internally.
 	toHTMLString() {
-		return (
-			this.originalHTML ||
-			toHTMLString( { value: this[ RichTextInternalData ] } )
-		);
+		return this.originalHTML || toHTMLString( { value: this.#value } );
```
Comment (jsnajdr) [inline packages/rich-text/src/create.js:148]: The way how `originalHTML` is defined in `fromHTMLElement` is not entirely correct. It effectively creates a _live binding_ between the `htmlElement` and the `RichTextData` value. If the element's contents change, then `toHTMLString()` will read the latest `.innerHTML` value in the getter.

Other parts of the `RichTextData` value are however not live: the `toPlainText` method, the `.text` getter, etc. The object's content can be inconsistent and out of sync.

Comment (ellatrix) [reply packages/rich-text/src/create.js:148]: RichTextData should not be mutable

**Comment by ellatrix** [comment]
I have tried this but it didn't work for me. Would be cool if it works :)

**Comment by jsnajdr** [comment]
Oh, TypeScript makes a complaint I don't understand:
```
error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.
```
because we define `target: 'esnext'` in our `tsconfig.json`.

File: packages/rich-text/src/create.js (hunk 1 above)
Comment (jsnajdr) [reply packages/rich-text/src/create.js:148]: Then we need to read `htmlElement.innerHTML` only once in `RichTextData` constructor, not on every call of `.toHTMLString`.

Comment (ellatrix) [reply packages/rich-text/src/create.js:148]: We do it only once in the static method? `originalHTML` is a value not a getter?

Comment (ellatrix) [reply packages/rich-text/src/create.js:148]: Or do you mean `toHTMLString()`? We shouldn't calculate it in the constructor for performance, we sure we could cache it here for the next call

Comment (ellatrix) [reply packages/rich-text/src/create.js:148]: (We only want to serialise when it's needed)

Comment (jsnajdr) [reply packages/rich-text/src/create.js:148]: > `originalHTML` is a value not a getter?

Oh, somehow I overlooked this 🙂 It's alright then.

**Review by ellatrix — APPROVED** [review]
Very nice :)

**Comment by jsnajdr** [comment]
I fixed the TypeScript error. It was the `check-build-type-declaration-files.js` script that runs `tsc` on individual files. It doesn't build/check a project that has a `tsconfig.json`, but individual files, and no config is loaded. I added the required `target` and `moduleResolution` options on command line.

Maybe @noahtallen or @sirreal know a way how to make `tsc` read a `tsconfig.json`.

**Comment by sirreal** [comment]
The goal is to run `tsc` on arbitrary files but with a provided compiler options from a tsconfig file? I believe `--project` and `--build` are the options to pass a tsconfig, but those will typically determine whats included in a compilation as well. I don't know whether you can use those options with a source file to process argument.

**Comment by jsnajdr** [comment]
> The goal is to run `tsc` on arbitrary files but with a provided compiler options from a tsconfig file?

Yes, the goal is to typecheck one specific file with a config for a particular project.

**Comment by borislavfil** [comment]
Hello, I am getting "SyntaxError: Invalid character: '#'" on Safari IOS 14.3 and I have narrowed it down to this MR. It seems to work correctly on any other browser I have tested (Windows and Android). I am not sure if I should create a new Issue

**Pull request merged**